### PR TITLE
BACK-1364: Integrate SwapBased on Base

### PR DESCRIPTION
**Pull request by JackieJoo**

### CODE REVIEW

File: src/dex/uniswap-v3/uniswap-v3-integration.test.ts [hunk 1]
```diff
@@ -1239,3 +1240,377 @@ describe('Retro', () => {
...
+
+describe('BaseswapV3', function () {
+  const dexKey = 'BaseswapV3';
+
+  describe('Base', function () {
+    const network = Network.BASE;
+    const dexHelper = new DummyDexHelper(network);
+
+    let blockNumber: number;
+    let retro: UniswapV3;
```
Comment (aburkut) [inline src/dex/uniswap-v3/uniswap-v3-integration.test.ts:1251]: @JackieJoo Let's rename retro -> baseSwapV3

File: src/dex/uniswap-v3/uniswap-v3-integration.test.ts [hunk 2]
```diff
@@ -1239,3 +1240,377 @@ describe('Retro', () => {
...
+    const TokenA = Tokens[network][TokenASymbol];
+
+    const TokenBSymbol = 'WETH';
+    const TokenB = Tokens[network][TokenBSymbol];
+
+    const QuoterV2 = '0x4fDBD73aD4B1DDde594BF05497C15f76308eFfb9';
+
+    beforeEach(async () => {
+      blockNumber = await dexHelper.web3Provider.eth.getBlockNumber();
+      retro = new UniswapV3(network, dexKey, dexHelper);
```
Comment (aburkut) [inline src/dex/uniswap-v3/uniswap-v3-integration.test.ts:1263]: @JackieJoo The same

File: src/dex/uniswap-v3/uniswap-v3-integration.test.ts [hunk 3]
```diff
@@ -1239,3 +1240,377 @@ describe('Retro', () => {
...
+    });
+
+    it('getPoolIdentifiers and getPricesVolume SELL', async function () {
+      const pools = await retro.getPoolIdentifiers(
```
Comment (aburkut) [inline src/dex/uniswap-v3/uniswap-v3-integration.test.ts:1267]: @JackieJoo The same

File: src/dex/uniswap-v3/uniswap-v3-integration.test.ts (hunk 1 above)
Comment (JackieJoo) [reply src/dex/uniswap-v3/uniswap-v3-integration.test.ts]: done

**Review by aburkut — APPROVED** [review]

File: src/dex/uniswap-v3/uniswap-v3-integration.test.ts [hunk 4]
```diff
@@ -1239,3 +1239,380 @@ describe('Retro', () => {
...
+    let blockNumber: number;
+    let baseswapV3: UniswapV3;
+
+    const TokenASymbol = 'USDC';
+    const TokenA = Tokens[network][TokenASymbol];
+
+    const TokenBSymbol = 'WETH';
+    const TokenB = Tokens[network][TokenBSymbol];
+
+    const QuoterV2 = '0x4fDBD73aD4B1DDde594BF05497C15f76308eFfb9';
```
Comment (paraschiv-razvan) [inline src/dex/uniswap-v3/uniswap-v3-integration.test.ts:1259]: Why do we not use the quoter address from the config?

File: src/dex/uniswap-v3/uniswap-v3-integration.test.ts [hunk 5]
```diff
@@ -1239,3 +1239,380 @@ describe('Retro', () => {
...
+    });
+
+    it('getPoolIdentifiers and getPricesVolume BUY stable pairs', async function () {
+      const TokenASymbol = 'DAI';
+      const TokenA = Tokens[network][TokenASymbol];
+
+      const TokenBSymbol = 'USDC';
+      const TokenB = Tokens[network][TokenBSymbol];
+
+      const amountsBuy = [
```
Comment (paraschiv-razvan) [inline src/dex/uniswap-v3/uniswap-v3-integration.test.ts:1497]: This is duplicated. Maybe it should be a good idea to have one variable.

**Review by paraschiv-razvan — APPROVED** [review]

File: src/dex/uniswap-v3/uniswap-v3-integration.test.ts (hunk 5 above)
Comment (JackieJoo) [reply src/dex/uniswap-v3/uniswap-v3-integration.test.ts]: done

**Pull request merged**